crates/oxc_linter/src/rules/react/jsx_fragments.rs

```diff
@@ -10,23 +10,26 @@ use crate::{AstNode, context::LintContext, rule::Rule, utils::is_jsx_fragment};
 
 fn jsx_fragments_diagnostic(span: Span, mode: FragmentMode) -> OxcDiagnostic {
     let msg = if mode == FragmentMode::Element {
-        "Standard form for React fragments is preferred"
+        "Standard form for React fragments is preferred."
     } else {
-        "Shorthand form for React fragments is preferred"
+        "Shorthand form for React fragments is preferred."
     };
     let help = if mode == FragmentMode::Element {
-        "Use <React.Fragment></React.Fragment> instead of <></>"
+        "Use `<React.Fragment></React.Fragment>` instead of `<></>`."
     } else {
-        "Use <></> instead of <React.Fragment></React.Fragment>"
+        "Use `<></>` instead of `<React.Fragment></React.Fragment>`."
     };
     OxcDiagnostic::warn(msg).with_help(help).with_label(span)
 }
 
-#[derive(Debug, Default, Clone, JsonSchema, Deserialize, Serialize)]
-#[serde(rename_all = "camelCase", default)]
+#[derive(Debug, Default, Clone)]
 pub struct JsxFragments {
-    /// `syntax` mode:
-    ///
+    mode: FragmentMode,
+}
```

Comment:
you can support both by doing an enum, something like this i think
```rs
#[schemars(untagged)]
pub enum JsxFragments {
  StringFragmentMode(FragmentMode);
  ObjectFragmentMode({ mode: FragmentMode });
}
```

Reply:
I've pushed up a commit to use this pattern, although I'm not really sure it's particularly better than the previous version. I'll probably let cam decide on whatever is preferred.

Reply:
imo it's better because it codifies the config in the type rather than some adhoc deserialization code

Reply:
Unfortunately we'll need to make it possible to serialize a value of this type for the website to generate it, right now it can't handle an enum that is more than just simple types.

Reply:
DefaultRuleConfig definitely makes it better: 65f83444aa3e908e5464dbd27ce53380e24c0a40